### PR TITLE
Update sync documentation

### PR DESCRIPTION
**Pull request by ErikSchierboom**
The current documentation for `tests.toml` files was still referring to the obsolete canonical data syncer. This PR updates the documentation to use configlet's `sync` command instead. It also updates the contents of the example `tests.toml` file to the latest spec.

Closes https://github.com/exercism/configlet/issues/124

### CODE REVIEW

File: README.md
Comment (ee7) [inline README.md:212]:
```suggestion
To download the latest version of the `configlet` tool, please run the [`fetch-configlet`](https://github.com/exercism/configlet/blob/main/scripts/fetch-configlet) script or the [`fetch-configlet.ps1`](https://github.com/exercism/configlet/blob/main/scripts/fetch-configlet.ps1) script. At least one of these scripts should already exist in every track repo's `bin` directory - the script will also download `configlet` to this location. You can then sync the tests by running `./bin/configlet sync` (on Linux or macOS) or `.\bin\configlet.exe sync` (on Windows).
```

Non-blocking suggestion.

This suggested wording could be improved, but the idea is that it's better to tell the user to run the one that's already in the track repo. If we tell them to download the fetch script themselves, the user still has to save it to the `bin` directory - most of the time this just means they'll be prompted about overwriting the existing one.

Referring to `configlet sync` as "the tool" also felt a bit awkward - I'd say that "the tool" is `configlet` itself.

File: README.md
Comment (ee7) [inline README.md:210]: Non-blocking comment: the default `sync` behavior will stop being interactive in the future. We should remember to update this later.

**Review by ee7 — APPROVED** [review]
LGTM.

I did some nitpicking, partly unrelated to the `canonical_data_syncer` to `configlet sync` change - but is still in the scope of "update sync documentation"

And I've checked that, with this PR, running
```
$ git grep 'syncer'
```
in this repo produces no output.

File: README.md
Comment (ErikSchierboom) [reply README.md:210]: We should

**Review by wolf99 — APPROVED** [review]

**Comment by ee7** [comment]
Pinging @exercism/reviewers for a review.

This is an important docs update - the current wording has probably already caused a lot of confusion.

**Comment by ErikSchierboom** [comment]
@SleeplessByte Care to review?

**Comment by ee7** [comment]
This PR should no longer be merged as-is - I'll update it to reflect [the latest changes](https://github.com/exercism/configlet/releases/tag/4.0.0-alpha.21) to `configlet sync`. A plain `configlet sync` is no longer interactive.

**Comment by ee7** [comment]
I've also added some extra docs.

In the future, it would be nice if we didn't have to maintain 3 sets of `configlet` docs (in `exercism/configlet`, `exercism/docs`, and `exercism/problem-specifications`).

**Comment by ErikSchierboom** [comment]
Thanks! Ping @exercism/reviewers

**Review by hiljusti — CHANGES_REQUESTED** [review]

**Comment by hiljusti** [comment]
Minor suggestions aside... lgtm

**Review by hiljusti — APPROVED** [review]

**Comment by ee7** [comment]
@hiljusti: Thanks for the review. You raise some good points about the wording, but it's a little tricky to find a great one.

Some things to consider:
1. A user can run a bash script on Windows, and when running on Windows, the `fetch-configlet` script correctly downloads the Windows release of `configlet`.
1. We might prefer to point Windows users towards the bash script because:
   1. The bash script is in every track repo, but the PowerShell script is only in 11 of 79 track repos. (The repos are: `cpp`, `csharp`, `dart`, `fsharp`, `javascript`, `red`, `solidity`, `typescript`, `wren`, `z3`, `zig`).
   1. Many Windows users are more familiar with bash than PowerShell.
   1. The bash script has been reviewed much more thoroughly than the PowerShell script.
1. A user can theoretically run a PowerShell script on a non-Windows OS. However, the `fetch-configlet.ps1` always downloads the Windows release of `configlet`, so we should indeed communicate "only run the `.ps1` script on Windows".
1. We currently only build `configlet` releases for Linux, macOS, and Windows (all `x86_64` only).
1. The `fetch-configlet` script does run without error on the BSDs (as long as bash is installed). It downloads the Linux binary.
1. The Linux binary works fully on FreeBSD and NetBSD if Linux binary compatibility is enabled, as described e.g. [here](https://docs.freebsd.org/en/books/handbook/linuxemu/). However, I have not documented this fact, and we don't test it during CI. But I **am** willing to document it - I think it will stay working.
1. The Linux binary is statically linked, which makes it easier to run on the BSDs that provide Linux binary compatibility.
1. On BSDs that don't provide Linux binary compatibility (e.g. OpenBSD) a user who wants to use `configlet` must compile it themselves, or run a release binary in a VM.
1. I have tested that `configlet` compiles and runs properly on at least FreeBSD, OpenBSD, and NetBSD.

@ErikSchierboom any thoughts on the wording here?

Also: in the future, can we have a single source of truth for `configlet` installation instructions? If so, should that be in `exercism/docs`, or `exercism/configlet`? Then the instructions in `exercism/problem-specifications` can be minimal, and point elsewhere for more details.

**Comment by hiljusti** [comment]
All sounds reasonable to me

**Comment by ErikSchierboom** [comment]
> Many Windows users are more familiar with bash than PowerShell.

In my experience, I've not found this to be true. Bash isn't even installed by default on Windows, so I don't think we can assume Windows users will just be able to run the Bash command.

> Also: in the future, can we have a single source of truth for configlet installation instructions? If so, should that be in exercism/docs, or exercism/configlet? Then the instructions in exercism/problem-specifications can be minimal, and point elsewhere for more details.

I think having the documentation be in exercism/configlet and referring to that from the docs and prob-specs repos makes the most sense.

File: README.md
Comment (ErikSchierboom) [reply README.md:201]: I like the suggestion. @ee7?

**Comment by hiljusti** [comment]
I'm fine with not further specifying OSs. It's kinda minor; I think Windows developers will recognize a `.ps1` as a powershell script, and non-windows developers will ignore it because it's foreign. BSD developers are used to software (outside of ports) not working out-of-the-box

Rambly thoughts follow:

Since this targets track maintainers, I think it's less important what the OS image comes with and more important what developers will have installed. I'm not a Windows expert, though I do have one machine that runs it, and I even ran a Windows machine for a few weeks recently for work... though I just used it as a shell to reach other remote hosts I could use for development

Anyway - I know Bash won't be installed on Windows by default, but what is the developer setup these days for a Windows dev?  Not all Linux distros will have `git` installed by default, but we expect people to have at least gotten that far if they're maintaining tracks. I thought the Cygwin/MinGW/GitBash days were mostly over since Windows Subsystem Linux 2.0 can be run as a (non-virtualized) peer to windows now. Not sure if that's what people are using, though. I've heard of some difficulties when you want to do deeper Linux networking things, since Windows can gate the Linux environment.

**Comment by ErikSchierboom** [comment]
> Since this targets track maintainers, I think it's less important what the OS image comes with and more important what developers will have installed. 

Correct, until we centralize the documentation in one place, in which case it will become aimed at students too.

> Anyway - I know Bash won't be installed on Windows by default, but what is the developer setup these days for a Windows dev?

Unless you're running things exclusively in WSL, Windows users will _probably_ have installed Git locally too. That means they'll have Git bash, but I never used in my 10+ years of running Git on Windows.

**Comment by SleeplessByte** [comment]
I use git bash all the time, but it's important to note that those bash scripts _only_ work in a git bash environment, which you're not always in.

I would prefer keeping both the "cross-env-if-you're-bash-like" and "always-works-on-windows" scripts.

File: README.md
Comment (kotp) [inline README.md:210]:
```suggestion
To make it easy to keep the `tests.toml` up to date, tracks should use the [`configlet` application](https://github.com/exercism/configlet)'s `sync` command.
```

Should we encourage this behavior?  So "should" instead of "can"?

Comment (ErikSchierboom) [reply README.md:210]: I think we should (pun intended)! @ee7?

**Review by ee7 — COMMENTED** [review]
I mostly liked @hiljusti's suggestion, I just wanted to:
- avoid the possible reading of "the bash script cannot be run on Windows"
- clarify that the PowerShell script is Windows-only

As a final thing before we merge this - how about this?

File: README.md
Comment (ErikSchierboom) [reply README.md:228]: I like it! In the future, we could consider modifying the PowerShell script to also download the binaries for non-Windows OS'es (PowerShell is cross-platform nowadays).

**Comment by ErikSchierboom** [comment]
Thanks everyone!

**Pull request merged**
